src/languageclient.ts

```diff
@@ -0,0 +1,202 @@
+        const executablePath = Path.join(serverLocation, executableName);
+
+        if (!fs.existsSync(executablePath)) {
+            let continueInstall = await vscode.window.showInformationMessage(
+                "Would you like to install Terraform Language Server from juliosueiras/terraform-lsp? \n This provides experimental Terraform 0.12 support",
+                "Install", "Abort");
+            if (continueInstall === "Abort") {
+                vscode.window.showErrorMessage("Terraform language server install aborted");
+                return;
+            }
+            await this.installLanguageServer(thisPlatform, serverLocation);
+        }
+
+        let serverOptions: ServerOptions = {
+            command: executablePath,
+            args: [],
+            options: {
+                env: {
+                    ...process.env
+                }
+            }
+        };
+
+        // Options to control the language client
+        let clientOptions: LanguageClientOptions = {
+            documentSelector: [{ scheme: 'file', language: 'terraform' }],
+            synchronize: {
+                configurationSection: 'terraformLanguageServer',
+                // Notify the server about file changes to '.clientrc files contained in the workspace
+                fileEvents: workspace.createFileSystemWatcher('**/*.tf'),
+            }
+        };
+
+        // Create the language client and start the client.
+        let langClient = new LanguageClient(
+            'terraformLanguageServer',
+            'Language Server',
+            serverOptions,
+            clientOptions,
+            true
+        );
+
+        langClient.trace = 2;
+
+        langClient.onReady().then(() => {
+            const capabilities = langClient.initializeResult && langClient.initializeResult.capabilities;
+            if (!capabilities) {
+                return vscode.window.showErrorMessage('The language server is not able to serve any features at the moment.');
+            } else {
+                vscode.window.showInformationMessage('Terraform Language server started.')
+            }
+        });
+
+        // Start the client. This will also launch the server
+        ctx.subscriptions.push(langClient.start());
+
+        // Register command to re-run installer
+        const commandHandler = async () => {
+            try {
+                await langClient.stop();
+            } catch {
+                // Expected
+            }
+            // Todo:
+            // Shutdown method on LSP not implemented :(
+            //    `await langClient.stop();` fails
+            //
+            // We need the process to be stopped so we can overwrite the binary with the new version.
+            // This is a hack for the time beind based on internal property of the languageClient object.
+            const PID = langClient["_serverProcess"].pid;
+            process.kill(PID, 9)
```

Comment:
To remove the need for this nasty hack I've created a PR on the LSP to add a `shutdown` command so `langClient.stop()` will work.